cms/djangoapps/contentstore/rest_api/v0/tests/test_tabs.py

```diff
@@ -0,0 +1,225 @@
+        self.test_tab = ItemFactory.create(
+            parent_location=self.course.location,
+            category="static_tab",
+            display_name="Static_1",
+        )
+        self.reload_course()
+
+    def check_invalid_response(self, resp):
+        """Verify response is an error listing the invalid_tab_id"""
+
+        assert resp.status_code, 400
+        resp_content = json.loads(resp.content.decode("utf-8"))
+        assert "developer_message" in resp_content
+        return resp_content["developer_message"]
+
+    def test_reorder_tabs(self):
+        """Test re-ordering of tabs"""
+
+        # get the original tab ids
+        orig_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        tab_ids = list(orig_tab_ids)
+        num_orig_tabs = len(orig_tab_ids)
+
+        # make sure we have enough tabs to play around with
+        assert num_orig_tabs >= 5
+
+        # reorder the last two tabs
+        tab_ids[num_orig_tabs - 1], tab_ids[num_orig_tabs - 2] = tab_ids[num_orig_tabs - 2], tab_ids[num_orig_tabs - 1]
+
+        # remove the middle tab
+        # (the code needs to handle the case where tabs requested for re-ordering is a subset of the tabs in the course)
+        removed_tab = tab_ids.pop(num_orig_tabs // 2)
+        assert len(tab_ids) == num_orig_tabs - 1
+
+        # post the request
+        resp = self.client.post(
+            self.url,
+            data={"tabs": [{"tab_id": tab_id} for tab_id in tab_ids]},
+            content_type="application/json",
+        )
+        assert resp.status_code == 204
+
+        # reload the course and verify the new tab order
+        self.reload_course()
+        new_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        assert new_tab_ids == tab_ids + [removed_tab]
+        assert new_tab_ids != orig_tab_ids
+
+    def test_reorder_tabs_invalid_list(self):
+        """Test re-ordering of tabs with invalid tab list"""
+
+        orig_tab_ids = [tab.tab_id for tab in self.course.tabs]
+        tab_ids = list(orig_tab_ids)
+
+        # reorder the first two tabs
+        tab_ids[0], tab_ids[1] = tab_ids[1], tab_ids[0]
+
+        # post the request
+        resp = self.client.post(
+            self.url,
+            data={"tabs": [{"tab_id": tab_id} for tab_id in tab_ids]},
+            content_type="application/json",
+        )
+        assert resp.status_code == 400
+        error = self.check_invalid_response(resp)
+        assert "error" in error
+
+    def test_reorder_tabs_invalid_tab(self):
+        """Test re-ordering of tabs with invalid tab"""
+
+        invalid_tab_ids = ["courseware", "info", "invalid_tab_id"]
+
+        # post the request
+        resp = self.client.post(
+            self.url,
+            data={"tabs": [{"tab_id": tab_id} for tab_id in invalid_tab_ids]},
+            content_type="application/json",
+        )
+        self.check_invalid_response(resp)
+
+    def check_toggle_tab_visibility(self, tab_type, new_is_hidden_setting):
```

Comment:
I think this helper is checking as well as updating the tab order, so for that instance, the name of the method should be updated.

Reply:
It is not changing the tab order, only the visibility.